mimic/rest/maas_api.py

```diff
@@ -642,10 +642,21 @@ def view_agent_host_info(self, request, tenant_id):
                                'details': 'You must specify an agentId, entityId, or an entity URI.',
                                'mimicNotes': 'But mimic will only accept entityId right now',
                                'txnId': '.fake.mimic.transaction.id.c-1111111.ts-123444444.v-12344frf'})
+
         entity_id = request.args['entityId'][0].strip()
         for e in self._entity_cache_for_tenant(tenant_id).entities_list:
             if e['id'] == entity_id:
                 agent_id = e['agent_id']
+                break
+        else:
```

Comment:
On second thought it is absolutely fine as is. The alternative way would be,
```
entities = [e for e in self._entity_cache_for_tenant(tenant_id).entities_list if e['id'] == entity_id]
if entities:
    agent_id = entities[0]['agent_id']
else:
    request.setResponseCode(404)
    ....
```
I would even consider having a function for the list comprehension as it seems like something that could be reused.